Makefile

```diff
@@ -141,7 +141,7 @@ lint: alloylint
 # final command runs tests for all other submodules.
 test:
 	$(GO_ENV) go test $(GO_FLAGS) -race $(shell go list ./... | grep -v /integration-tests/)
-	$(GO_ENV) go test $(GO_FLAGS) ./internal/static/integrations/node_exporter ./internal/static/logs ./internal/component/otelcol/processor/tail_sampling ./internal/component/loki/source/file ./internal/component/loki/source/docker ./internal/component/prometheus/remote/queue/serialization
+	$(GO_ENV) go test $(GO_FLAGS) ./internal/static/integrations/node_exporter ./internal/static/logs ./internal/component/otelcol/processor/tail_sampling ./internal/component/loki/source/file ./internal/component/loki/source/docker ./internal/component/prometheus/remote/queue/serialization  ./internal/component/prometheus/remote/queue/network
```

Comment:
Because I am reaching into the config to test for updating config which happens on a different thread than created it so the race detector is unhappy. I thought about making the ability to retrieve safely but was a lot of effort for one test for something unused in actual code.

Reply:
The manager code itself would need synchronization primitives in addition to the test which is not ideal. What do you mean by using function pointers?

Reply:
Resolved this by ensuring the records coming in are written by the new batch size.